src/building/model.c

```diff
@@ -193,199 +205,158 @@ int model_load(void)
-        case BUILDING_SHRINE_MERCURY:
-        case BUILDING_SHRINE_NEPTUNE:
-        case BUILDING_SHRINE_VENUS:
-            return &MODEL_LARARIUM;
-        case BUILDING_NYMPHAEUM:
-            return &MODEL_NYMPHAEUM;
-        case BUILDING_WATCHTOWER:
-            return &MODEL_WATCHTOWER;
-        case BUILDING_SMALL_MAUSOLEUM:
-            return &MODEL_SMALL_MAUSOLEUM;
-        case BUILDING_LARGE_MAUSOLEUM:
-            return &MODEL_LARGE_MAUSOLEUM;
-        case BUILDING_CARAVANSERAI:
-            return &MODEL_CARAVANSERAI;
-        case BUILDING_PALISADE:
-        case BUILDING_PALISADE_GATE:
-            return &MODEL_PALISADE;
-        case BUILDING_HIGHWAY:
-            return &MODEL_HIGHWAY;
-        case BUILDING_GOLD_MINE:
-            return &MODEL_GOLD_MINE;
-        case BUILDING_STONE_QUARRY:
-            return &MODEL_STONE_QUARRY;
-        case BUILDING_SAND_PIT:
-            return &MODEL_SAND_PIT;
-        case BUILDING_BRICKWORKS:
-            return &MODEL_BRICKWORKS;
-        case BUILDING_CONCRETE_MAKER:
-            return &MODEL_CONCRETE_MAKER;
-        case BUILDING_CITY_MINT:
-            return &MODEL_CITY_MINT;
-        case BUILDING_DEPOT:
-            return &MODEL_DEPOT;
-        case BUILDING_OVERGROWN_GARDENS:
-            return &buildings[BUILDING_GARDENS];
-        case BUILDING_ARMOURY:
-            return &MODEL_ARMOURY;
-        case BUILDING_LATRINES:
-            return &MODEL_LATRINE;
-        case BUILDING_NATIVE_DECORATION:
-            return &MODEL_NATIVE_DECORATION;
-        case BUILDING_NATIVE_MONUMENT:
-            return &MODEL_NATIVE_MONUMENT;
-        case BUILDING_NATIVE_WATCHTOWER:
-            return &MODEL_NATIVE_WATCHTOWER;
-        default:
-            break;
+    if (type > NUM_BUILDINGS + NUM_NEW_BUILDINGS) {
+        return &NOTHING;
     }
-
-    if ((type >= BUILDING_PINE_TREE && type <= BUILDING_SENATOR_STATUE) ||
-        type == BUILDING_HEDGE_DARK || type == BUILDING_HEDGE_LIGHT ||
-        type == BUILDING_DECORATIVE_COLUMN || type == BUILDING_LOOPED_GARDEN_WALL ||
-        type == BUILDING_COLONNADE || type == BUILDING_LOOPED_GARDEN_WALL ||
-        type == BUILDING_ROOFED_GARDEN_WALL || type == BUILDING_GARDEN_PATH ||
-        type == BUILDING_PANELLED_GARDEN_WALL || type == BUILDING_GLADIATOR_STATUE) {
-        return &buildings[BUILDING_SMALL_STATUE];
-    }
-
-    if (type == BUILDING_SMALL_POND || type == BUILDING_OBELISK ||
-        type == BUILDING_LEGION_STATUE || type == BUILDING_DOLPHIN_FOUNTAIN) {
-        return &buildings[BUILDING_MEDIUM_STATUE];
-    }
-
-    if (type == BUILDING_LARGE_POND || type == BUILDING_HORSE_STATUE) {
-        return &buildings[BUILDING_LARGE_STATUE];
-    }
-
-    if (type == BUILDING_FORT_AUXILIA_INFANTRY || type == BUILDING_FORT_ARCHERS) {
-        return &buildings[BUILDING_FORT_LEGIONARIES];
-    }
-
-    if (type > 129) {
-        return &MODEL_NULL;
+    if (type > 114) {
+        return &buildings[type + 15];
```

Comment:
If you do as I said above, this becomes useless and you can always return `&buildings[type]`